File: 09_13_2024_XOR_queries_of_subarray.py

class Solution:
    def xorQueries(self, arr, queries):
        temp = [0] * len(arr)
        temp[0] = arr[0]

        for i in range(1, len(arr)):
            temp[i] = temp[i - 1] ^ arr[i]

        ans = []
        for a, b in queries:
            if a > 0:
                ans.append(temp[b] ^ temp[a - 1])
            else:
                ans.append(temp[b])

        return ans
        

File: 09_13_2024_XOR_queries_of_subarray.java

class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int[] temp = new int[arr.length];
        temp[0] = arr[0];

        for (int i = 1; i < arr.length; i++) {
            temp[i] = temp[i - 1] ^ arr[i];
        }

        int[] ans = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int a = queries[i][0], b = queries[i][1];
            if (a > 0) {
                ans[i] = temp[b] ^ temp[a - 1];
            } else {
                ans[i] = temp[b];
            }
        }

        return ans;
    }
}
